s3/s3.go

```diff
@@ -829,13 +832,23 @@ func (o *Object) Hash(t fs.HashType) (string, error) {
 	if t != fs.HashMD5 {
 		return "", fs.ErrHashUnsupported
 	}
-	etag := strings.Trim(strings.ToLower(o.etag), `"`)
+	hash := strings.Trim(strings.ToLower(o.etag), `"`)
```

Comment:
I switched to the more general `hash` name.  I think `etag` was too specific because the variable sometimes holds an MD5 sum.